server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -904,15 +907,24 @@ private static Mapper getMapper(final ParseContext context, ObjectMapper objectM
             }
         }
         String leafName = subfields[subfields.length - 1];
-        mapper = objectMapper.getMapper(leafName);
+        return objectMapper.getMapper(leafName);
+    }
+
+    private static Mapper getLeafMapper(final ParseContext context,
```

Comment:
I can't seem to find a name that clearly explains why we have this new method as opposed to the existing getMapper one. Shall we add a comment to explain what is the difference between the two and why?